test/integration/Elsa.Workflows.IntegrationTests/Scenarios/RunAsynchronousActivityOutput/Tests.cs

```diff
@@ -0,0 +1,164 @@
+        Assert.Equal(32, activityExecutionRecord.Outputs!.GetValue<int>("Product"));
+
+        var activityOutputRegister = workflowFinishedRecord.WorkflowExecutionContext.GetActivityOutputRegister();
+        Assert.Equal(2, activityOutputRegister.FindMany(p => true).Count());
+        Assert.Equal(12, activityOutputRegister.FindOutputByActivityId("SampleActivity1", "Sum"));
+        Assert.Equal(32, activityOutputRegister.FindOutputByActivityId("SampleActivity1", "Product"));
+    }
+
+    [Theory(DisplayName = "Activity outputs captured in activity execution record")]
+    [InlineData(true)]
+    [InlineData(false)]
+    public async Task ActivityOutputCaptureParallelTest(bool runAsynchronously)
+    {
+
+        // Arrange
+
+        TestWorkflow workflow = new(workflowBuilder =>
+            {
+                workflowBuilder.DefinitionId = "sampleWorkflow";
+                workflowBuilder.Id = "sampleWorkflow";
+
+                var variable1 = new Variable<int>();
+                var variable2 = new Variable<int>();
+                workflowBuilder.Root = new Elsa.Workflows.Activities.Parallel()
+                {
+                    Variables =
+                        {
+                            variable1,
+                            variable2,
+                        },
+                    Activities =
+                        {
+                            new SampleActivity()
+                            {
+                                Id = "SampleActivity1",
+                                RunAsynchronously = runAsynchronously,
+                                Number1 = new(4),
+                                Number2 = new(8),
+                                Sum = new(variable1),
+                            },
+                            new SampleActivity()
+                            {
+                                Id = "SampleActivity2",
+                                RunAsynchronously = runAsynchronously,
+                                Number1 = new(2),
+                                Number2 = new(7),
+                                Product = new(variable2),
+                            }
+                        }
+                };
+            }
+        );
+
+        var activityExecutionStore = new MemoryActivityExecutionStore(new MemoryStore<ActivityExecutionRecord>());
+
+        // Act
+
+        WorkflowFinished? workflowFinishedRecord = await workflow.DispatchWorkflowAndRunToCompletion(
+            configureServices: services =>
+            {
+                services.AddScoped<DistributedWorkflowRuntime>();
+            },
+            configureElsa: elsa =>
+            {
+                elsa.UseWorkflowRuntime(workflowRuntime => {
+                    workflowRuntime.ActivityExecutionLogStore = sp => activityExecutionStore;
+                    workflowRuntime.WorkflowRuntime = sp => sp.GetRequiredService<DistributedWorkflowRuntime>();

```

Comment:
Not sure if this is a bug or not... but to be able to run multiple asynchronous activities in parallel (the only way it makes sense to me) the new DistributedWorkflowRuntime workflow runtime must be used. I don't believe that this was not the case with 3.2.

Reply:
This was indeed intentional to require the `DistributedWorkflowRuntime` or `ProtoActorWorkflowRuntime` to have support for parallel activity execution. To be honest, I'd love it to be able to still have support for parallel activity execution using the `LocalWorkflowRuntime`. Originally, the `DistributedWorkflowRuntime` was created for hosting Elsa on multiple nodes, which requires distributed locking. But it's fair to expect that when running on a single node, you should still be able to run activities in parallel. Perhaps this could be added easily by updating the `LocalWorkflowRuntime` with a semaphore to synchronise access to the workflow instance.